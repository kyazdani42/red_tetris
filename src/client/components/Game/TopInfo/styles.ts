import styled from 'styled-components';

export const LeaveButtonStyle = styled.div`
  position: absolute;
  right: 4%;
  height: 30px;
  width: 70px;
  line-height: 30px;
  text-align: center;
  background-color: ${props => props.theme.colors.primary.dark};
  border: 1px solid rgba(255,255,255,0.2);
  box-shadow: 0 3px 6px rgba(255,255,255,0.05), 0 3px 6px rgba(255,255,255,0.13);
  transition: box-shadow 300ms;
  user-select: none;
  cursor: pointer;
  &:hover {
    box-shadow: 0 3px 6px rgba(255,255,255,0.2), 0 3px 6px rgba(255,255,255,0.3);
  };
`;

export const SpectreStyle = styled.div`
  border: 1px solid #fff;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 50px;
  height: 100px;
`;

export const SpectreBlockStyle = styled.div`
  width: 5px;
  height: 5px;
  background-color: ${(props: { col: number | null }) => props.col ? 'red' : 'black'};
`;

export const SpectresContainer = styled.div`
  width: 80%;
  overflow-x: auto;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
`;
