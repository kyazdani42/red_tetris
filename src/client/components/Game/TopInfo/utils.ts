export const getSpectreArray = (spectre: number[]) =>
  [...Array(20)].map((_, row) =>
    [...Array(10)].map((__, col) => (spectre[col] !== null && spectre[col] <= row ? 1 : null))
  );
