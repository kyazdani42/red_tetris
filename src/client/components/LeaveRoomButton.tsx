import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { leaveRoom } from '../actions/actions';

interface Props {
  dispatchLeaveRoom: () => void;
}

const Button = styled.div`
  height: 50px;
  width: 50px;
  background-color: ${props => props.theme.colors.primary.dark};
  border: 1px solid rgba(0,0,0,0.2);
  font-size: 17px;
  box-shadow: 0 3px 6px rgba(0,0,0,0.05), 0 3px 6px rgba(0,0,0,0.13);
  padding-top: 1px;
  padding-left: 5px;
  margin-left: 3px;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
  align-content: center;
  transition: box-shadow 300ms;
  user-select: none;
  cursor: pointer;
  &:hover {
    box-shadow: 0 3px 6px rgba(0,0,0,0.2), 0 3px 6px rgba(0,0,0,0.3);
  };
`;

const LeaveRoomButton: React.SFC<Props> = ({ dispatchLeaveRoom }) => (
  <Button onClick={dispatchLeaveRoom}>X</Button>
);

const mapDispatchToProps = (dispatch: any) => ({
  dispatchLeaveRoom: () => dispatch(leaveRoom())
});

export default connect(undefined, mapDispatchToProps)(LeaveRoomButton);
