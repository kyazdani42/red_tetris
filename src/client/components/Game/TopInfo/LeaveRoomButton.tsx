import * as React from 'react';
import { connect } from 'react-redux';

import { leaveRoom } from '../../../redux/actions';
import { LeaveButtonStyle } from './styles';

interface Props {
  dispatchLeaveRoom: () => void;
}

export const LeaveRoomButton: React.SFC<Props> = ({ dispatchLeaveRoom }) => (
  <LeaveButtonStyle onClick={dispatchLeaveRoom}>Quit</LeaveButtonStyle>
);

export const mapDispatchToProps = (dispatch: any) => ({
  dispatchLeaveRoom: () => dispatch(leaveRoom())
});

export default connect(undefined, mapDispatchToProps)(LeaveRoomButton);
