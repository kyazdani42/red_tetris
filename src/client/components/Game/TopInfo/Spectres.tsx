import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../../../redux/store';
import { NameStyle, SpectreBlockStyle, SpectresContainer, SpectreStyle, SpectreWrapper } from './styles';
import { getSpectreArray } from './utils';

interface Props {
  data: GameProps | null;
}

const Spectres: React.SFC<Props> = ({ data }) => {
  if (!data) return null;
  const spectres = getSpectres(data.otherPlayers);
  return <SpectresContainer>{spectres}</SpectresContainer>;
};

const getSpectres = (otherPlayers: GameProps['otherPlayers'] = []) =>
  otherPlayers.map((player, i) => (
    <SpectreWrapper key={`spectre-nb-${i}`}>
      <SpectreStyle>
        {getSpectreArray(player.spectre).map((row, j) => row.map((col, k) => (
          <SpectreBlockStyle
            col={col}
            key={`spectre-block-${j}-${k}`}
          />
        )))}
      </SpectreStyle>
      <NameStyle>{player.name}</NameStyle>
    </SpectreWrapper>
  ));

const mapStateToProps = (state: State): Props => ({
  data: state.app.gameData
});

export default connect(mapStateToProps)(Spectres);
