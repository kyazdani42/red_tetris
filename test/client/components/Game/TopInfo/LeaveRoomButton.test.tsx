// tslint:disable
import * as React from 'react';
import { shallow } from 'enzyme';
import { LeaveRoomButton, mapDispatchToProps } from '../../../../../src/client/components/Game/TopInfo/LeaveRoomButton';

describe('testing the component', () => {
  const props = {
    dispatchLeaveRoom: jest.fn()
  };
  const wrapper = shallow(<LeaveRoomButton {...props} />)
  it('smoke tests the component', () => {
    expect(wrapper.update()).toMatchSnapshot();
  })
  it('tests the onclick event', () => {
    wrapper.simulate('click');
    expect(props.dispatchLeaveRoom).toHaveBeenCalled();
  })
});

describe('testing mapDispatchToProps', () => {
  const dispatch = jest.fn();
  it('checks that dispatch is called', () => {
    mapDispatchToProps(dispatch).dispatchLeaveRoom();
    expect(dispatch).toHaveBeenCalled();
  })
})
