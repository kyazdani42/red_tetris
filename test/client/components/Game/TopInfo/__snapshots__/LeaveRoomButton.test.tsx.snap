// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the component smoke tests the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dispatchLeaveRoom={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Quit",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Quit",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  position: absolute;
  right: 4%;
  height: 30px;
  width: 70px;
  line-height: 30px;
  text-align: center;
  background-color: ",
          [Function],
          ";
  border: 1px solid rgba(255,255,255,0.2);
  box-shadow: 0 3px 6px rgba(255,255,255,0.05), 0 3px 6px rgba(255,255,255,0.13);
  transition: box-shadow 300ms;
  user-select: none;
  cursor: pointer;
  &:hover {
    box-shadow: 0 3px 6px rgba(255,255,255,0.2), 0 3px 6px rgba(255,255,255,0.3);
  };
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Quit",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Quit",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  position: absolute;
  right: 4%;
  height: 30px;
  width: 70px;
  line-height: 30px;
  text-align: center;
  background-color: ",
            [Function],
            ";
  border: 1px solid rgba(255,255,255,0.2);
  box-shadow: 0 3px 6px rgba(255,255,255,0.05), 0 3px 6px rgba(255,255,255,0.13);
  transition: box-shadow 300ms;
  user-select: none;
  cursor: pointer;
  &:hover {
    box-shadow: 0 3px 6px rgba(255,255,255,0.2), 0 3px 6px rgba(255,255,255,0.3);
  };
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
